### PR TITLE
feat(Button): add advanced handling of disabled prop

### LINKED ISSUE #1775: Button: the disabled attribute doesn't apply to the button element
### Steps
Create a `Button` component with disabled set to true.

### Expected Result
The button should be disabled.

### Actual Result
The button only has styles to make it appear disabled, but it still functions if you tab to it due to a missing disabled attribute on the button element.

### Version
0.68.3

### Testcase
https://codepen.io/anon/pen/OgbeXr

### PR DESCRIPTION
**Pull request by layershifter**
This PR:
- fixes #1775
- performs a refactoring of  the `render()` method
- improves the test coverage

### CODE REVIEW

File: test/specs/elements/Button/Button-test.js [hunk 1]
```diff
@@ -92,12 +121,19 @@ describe('Button', () => {
       shallow(<Button icon='user' />)
         .should.have.className('icon')
     })
+
+    it('adds className icon when true', () => {
+      shallow(<Button icon />)
+        .should.have.className('icon')
+    })
```
Comment (layershifter) [inline test/specs/elements/Button/Button-test.js:128]: Missing test

File: src/elements/Button/Button.js [hunk 2]
```diff
@@ -256,56 +253,40 @@ class Button extends Component {
...
-    }
-
-    const labelElement = Label.create(label, { defaultProps: {
-      basic: true,
-      pointing: labelPosition === 'left' ? 'right' : 'left',
-    } })
-
-    if (labelElement) {
-      const classes = cx('ui', baseClasses, 'button', className)
+    if (!_.isNil(label)) {
```
Comment (layershifter) [inline src/elements/Button/Button.js:256]: If you take attention to these conditions you can see, that in fact there are two conditions: when the `Button` has a container and has not.

File: src/elements/Button/Button.js [hunk 3]
```diff
@@ -256,56 +253,40 @@ class Button extends Component {
...
-        </ElementType>
-      )
-    }
-
-    const labelElement = Label.create(label, { defaultProps: {
-      basic: true,
-      pointing: labelPosition === 'left' ? 'right' : 'left',
-    } })
-
-    if (labelElement) {
```
Comment (layershifter) [inline src/elements/Button/Button.js:275]: Also, this condition doesn't makes sence, we should use there check for the `label` prop.

File: src/elements/Button/Button.js [hunk 4]
```diff
@@ -256,56 +253,40 @@ class Button extends Component {
...
+      const labelElement = Label.create(label, { defaultProps: {
+        basic: true,
+        pointing: labelPosition === 'left' ? 'right' : 'left',
+      } })
 
       return (
         <ElementType {...rest} className={containerClasses} onClick={this.handleClick}>
           {labelPosition === 'left' && labelElement}
-          <button className={classes} ref={this.handleRef} tabIndex={tabIndex}>
+          <button className={buttonClasses} disabled={disabled} ref={this.handleRef} tabIndex={tabIndex}>
```
Comment (layershifter) [inline src/elements/Button/Button.js:267]: HTML's `button` tag [can have](https://developer.mozilla.org/en/docs/Web/HTML/Element/button) a `disabled` attr.

**Comment by levithomason** [comment]
Perfect, thank you.

**Comment by levithomason** [comment]
Released in `semantic-ui-react@0.69.0`

**Pull request merged**